### PR TITLE
Replace our own integer types with standard ones

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

File: src/openrct2/common.h [hunk 1]
```diff
@@ -47,17 +38,17 @@ using utf16string = utf16*;
...
-#define ror32(x, shift)     (((uint32)(x) >> (shift)) | ((uint32)(x) << (32 - (shift))))
-#define rol64(x, shift)     (((uint64)(x) << (shift)) | ((uint32)(x) >> (64 - (shift))))
-#define ror64(x, shift)     (((uint64)(x) >> (shift)) | ((uint32)(x) << (64 - (shift))))
+#define rol8(x, shift)      (((uint8_t)(x) << (shift)) | ((uint8_t)(x) >> (8 - (shift))))
+#define ror8(x, shift)      (((uint8_t)(x) >> (shift)) | ((uint8_t)(x) << (8 - (shift))))
+#define rol16(x, shift)     (((uint16_t)(x) << (shift)) | ((uint16_t)(x) >> (16 - (shift))))
+#define ror16(x, shift)     (((uint16_t)(x) >> (shift)) | ((uint16_t)(x) << (16 - (shift))))
+#define rol32(x, shift)     (((uint32_t)(x) << (shift)) | ((uint32_t)(x) >> (32 - (shift))))
+#define ror32(x, shift)     (((uint32_t)(x) >> (shift)) | ((uint32_t)(x) << (32 - (shift))))
+#define rol64(x, shift)     (((uint64_t)(x) << (shift)) | ((uint32_t)(x) >> (64 - (shift))))
```
Comment (Gymnasiast) [inline src/openrct2/common.h]: This looks like a copy-paste error to me (not introduced by this PR, mind you).

File: src/openrct2/config/Config.cpp [hunk 2]
```diff
@@ -142,47 +142,47 @@ namespace Config
         {
             auto model = &gConfigGeneral;
             model->always_show_gridlines = reader->GetBoolean("always_show_gridlines", false);
-            model->autosave_frequency = reader->GetSint32("autosave", AUTOSAVE_EVERY_5MINUTES);
+            model->autosave_frequency = reader->Getint32_t("autosave", AUTOSAVE_EVERY_5MINUTES);
```
Comment (Gymnasiast) [inline src/openrct2/config/Config.cpp]: Method names need to be fixed :P

File: src/openrct2/localisation/ConversionTables.cpp [hunk 3]
```diff
@@ -143,7 +143,7 @@ const encoding_convert_entry RCT2ToUnicodeTable[256] =
     { 124, FORMAT_INT32 },
     { 125, FORMAT_COMMA2DP32 },
     { 126, FORMAT_COMMA16 },
-    { 127, FORMAT_UINT16 },
+    { 127, FORMAT_uint16_t },
```
Comment (Broxzier) [inline src/openrct2/localisation/ConversionTables.cpp]: Unintended change

File: src/openrct2/localisation/FormatCodes.h [hunk 4]
```diff
@@ -58,7 +58,7 @@ enum {
     FORMAT_INT32,
     FORMAT_COMMA2DP32,
     FORMAT_COMMA16,
-    FORMAT_UINT16,
+    FORMAT_uint16_t,
```
Comment (Broxzier) [inline src/openrct2/localisation/FormatCodes.h]: Same here

**Comment by Broxzier** [comment]
I Couldn't request changes because it was merged in the meantime already, so I left them as individual comments now. Most of it looks good though, only one enum and two strings that were affected.

There was another one in `IniWriterTest.cpp`, though it doesn't really matter there, since both the input and expected strings were changed.

**Pull request merged**